document/src/tests/documentselectparsertest.cpp

```diff
@@ -182,11 +182,11 @@ DocumentSelectParserTest::createDocs()
         ArrayFieldValue abval(_doc.back()->getField("structarray").getDataType());
         {
             StructFieldValue sval1(aval.getNestedType());
-            sval1.set("key", 17);
-            sval1.set("value", "structval3");
+            sval1.setValue("key", std::make_unique<IntFieldValue>(17));
+            sval1.setValue("value", std::make_unique<StringFieldValue>("structval3"));
             StructFieldValue sval2(aval.getNestedType());
-            sval2.set("key", 18);
-            sval2.set("value", "structval4");
+            sval2.setValue("key", std::make_unique<IntFieldValue>(18));
+            sval2.setValue("value", std::make_unique<StringFieldValue>("structval4"));
```

Comment:
I wonder if we should have static factory functions (`IntFieldValue::make/create` etc) for creating these instead of explicit `make_unique`s everywhere. Thoughts?

Reply:
Agree, will do.